fairworkflows/fairstep.py

```diff
@@ -42,55 +47,6 @@ def from_rdf(cls, rdf, uri=None):
         self.anonymise_rdf()
         return self
 
-    @classmethod
-    def from_nanopub(cls, uri):
-        """
-            Fetches the nanopublication corresponding to the specified URI, and attempts to extract the rdf describing a fairstep from
-            its assertion graph. If the URI passed to this function is the uri of the nanopublication (and not the step itself) then
-            an attempt will be made to identify what the URI of the step actually is, by checking if the nanopub npx:introduces a
-            particular concept.
-
-            If the assertion graph does not contain any triples with the step URI as subject, an exception is raised. If such triples
-            are found, then ALL triples in the assertion are added to the rdf graph for this FairStep.
-        """
-        # Work out the nanopub URI by defragging the step URI
-        nanopub_uri, frag = urldefrag(uri)
-
-        # Fetch the nanopub
-        client = NanopubClient()
-        np = client.fetch(nanopub_uri)
-
-        # If there was no fragment in the original uri, then the uri was already the nanopub one.
-        # Try to work out what the step's URI is, by looking at what the np is introducing.
-        if len(frag) == 0:
-            concepts_introduced = []
-            for s, p, o in np.pubinfo.triples((None, namespaces.NPX.introduces, None)):
-                concepts_introduced.append(o)
-
-            if len(concepts_introduced) == 0:
-                raise ValueError('This nanopub does not introduce any concepts. Please provide URI to the step itself (not just the nanopub).')
-            elif len(concepts_introduced) > 0:
-                step_uri = str(concepts_introduced[0])
-
-            print('Assuming step URI is', step_uri)
-
-        else:
-            step_uri = uri
-        self = cls(uri=step_uri)
-
-        # Check that the nanopub's assertion actually contains triples refering to the given step's uri
```

Comment:
We don't raise ValueError anymore in this case. Instead we rely on the already existing warning raised here: https://github.com/fair-workflows/fairworkflows/blob/e3b78604b8ba663733ef9fbce37b00d5b30dd483/fairworkflows/fairstep.py#L45

I don't think it makes sense to have a warning in `from_rdf` and an error in `from_nanopub`, so I went for just raising a warning. Do you agree?

Reply:
Actually, now I am wondering if they shouldn't perhaps both be errors.  We could have a ```force``` kwarg to force it to accept anything if needed, but I think it might perhaps be less user friendly to accept rdf that contains no triples directly referencing the thing in question. The error message could suggest using ```force=True``` if the user really wants to ignore this?

Reply:
Good idea! I implemented it :)